validphys2/src/validphys/lhapdfset.py

```diff
@@ -0,0 +1,134 @@
+"""
+    Module containing an LHAPDF class compatible with validphys
+    using the official lhapdf python interface.
+
+    It exposes an interface mostly compatible with libNNPDF::LHAPDFSet
+    so it can be used as a drop-in replacement.
+
+    The ``.members`` and ``.central_member`` of the ``LHAPDFSet`` are
+    LHAPDF objects (the typical output from ``mkPDFs``) and can be used normally.
+    For MC PDFs the ``central_member`` is the average of all replicas (members 1-100)
+    while for Hessian PDFfs the ``central_member`` is also ``members[0]``
+
+    Examples
+    --------
+    >>> from validphys.lhapdfset import LHAPDFSet
+    >>> pdf = LHAPDFSet("NNPDF40_nnlo_as_01180", "replicas")
+    >>> len(pdf.members)
+    100
+    >>> pdf.central_member.alphasQ(91.19)
+    0.11800
+    >>> pdf.members[0].xfxQ2(0.5, 15625)
+    {-5: 6.983360500601136e-05,
+    -4: 0.0021818063617227604,
+    -3: 0.00172453472243952,
+    -2: 0.0010906577230485718,
+    -1: 0.0022049272225017286,
+    1: 0.020051104853608722,
+    2: 0.0954139944889494,
+    3: 0.004116641378803191,
+    4: 0.002180124185625795,
+    5: 6.922722705177504e-05,
+    21: 0.007604124516892057}
+"""
+import logging
+import numpy as np
+import lhapdf
+
+log = logging.getLogger(__name__)
+
+
+class LHAPDFSet:
+    """Wrapper for the lhapdf python interface.
+
+    Once instantiated this class will load the PDF set from LHAPDF.
+    If it is a T0 set only the CV will be loaded.
+
+    For Monte Carlo sets the central value (member 0) is by default not included when taking
+    the results for all members (i.e., when using ``grid_values``).
+    """
+
+    def __init__(self, name, error_type):
+        log.info("PDF: %s ErrorType: %s", name, error_type)
+        self._name = name
+        self._error_type = error_type
+        if self.is_t0:
+            # If at this point we already know this is a T0 set, load only the CV
+            self._lhapdf_set = [lhapdf.mkPDF(name)]
+        else:
+            self._lhapdf_set = lhapdf.mkPDFs(name)
+        self._flavors = None
+
+    @property
+    def is_monte_carlo(self):
```

Comment:
I am not sure if it is worth adding these seeing as there doesn't seem to be a strong case for them anywhere. For example `check_pdf_is_montecarlo` is indeed going to keep looking at `PDF` rather than `LHAPDFSet`

Reply:
Yes, it's useless. Actually in the 101-to-100 PR I removed it. I might as well remove it from here directly.